src/lib/verifyAppleIdToken.ts

```diff
@@ -1,51 +1,52 @@
-import * as jwt from 'jsonwebtoken';
-import * as jwksClient from 'jwks-rsa';
+import * as jwt from "jsonwebtoken";
+import * as jwksClient from "jwks-rsa";
+import { VerifyAppleIdTokenParams, VerifyAppleIdTokenResponse } from "./types";
 
-export const APPLE_BASE_URL = 'https://appleid.apple.com';
-export const JWKS_APPLE_URI = '/auth/keys';
+export const APPLE_BASE_URL = "https://appleid.apple.com";
+export const JWKS_APPLE_URI = "/auth/keys";
 
-export interface VerifyAppleIdTokenParams {
-  idToken: string;
-  clientId: string | string[];
-  nonce?: string;
-}
-
-export const getApplePublicKey = async (kid) => {
+export const getApplePublicKey = async (kid: string) => {
   const client = jwksClient({
     cache: true,
     jwksUri: `${APPLE_BASE_URL}${JWKS_APPLE_URI}`,
   });
-  const key: any = await new Promise((resolve, reject) => {
+  const key = await new Promise<jwksClient.SigningKey>((resolve, reject) => {
     client.getSigningKey(kid, (error, result) => {
       if (error) {
         return reject(error);
       }
       return resolve(result);
     });
   });
-  return (key.publicKey as jwksClient.CertSigningKey) || (key.rsaPublicKey as jwksClient.RsaSigningKey);
+  return key.getPublicKey();
 };
 
-export default async (params: VerifyAppleIdTokenParams) => {
+export async function verifyToken(params: VerifyAppleIdTokenParams) {
```

Comment:
Please do not change the type of the function :pray: I think all of the functions there are arrow functions, so keep it please.

Reply:
> arrow function is [a function expression, not a function declaration](https://stackoverflow.com/q/336859/1048572)

but how do you say 😀